integrations/object_store/src/backend/writer.rs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::borrow::Cow;
+use std::future::Future;
+use std::sync::Arc;
+
+use futures::FutureExt;
+use object_store::Attribute;
+use object_store::AttributeValue;
+use object_store::ObjectStore;
+use object_store::PutOptions;
+use object_store::PutPayload;
+use object_store::PutResult;
+
+use opendal::raw::*;
+use opendal::*;
+
+use super::error::parse_error;
+
+pub struct ObjectStoreWriter {
+    store: Arc<dyn ObjectStore + 'static>,
+    path: object_store::path::Path,
+    args: OpWrite,
+    result: Option<PutResult>,
+}
+
+impl ObjectStoreWriter {
+    pub fn new(store: Arc<dyn ObjectStore + 'static>, path: &str, args: OpWrite) -> Self {
+        Self {
+            store,
+            path: object_store::path::Path::from(path),
+            args,
+            result: None,
+        }
+    }
+}
+
+impl oio::Write for ObjectStoreWriter {
+    fn write(&mut self, bs: Buffer) -> impl Future<Output = Result<()>> + MaybeSend {
+        async move {
+            let bytes = bs.to_bytes();
+            let payload = PutPayload::from(bytes);
+            let opts = parse_write_args(&self.args)?;
+            let result = self
+                .store
+                .put_opts(&self.path, payload, opts)
```

Comment:
I think we can integrate with `MultipartWrite`

Reply:
updated the pr with MultipartWrite support, PTAL 🫡